napari/components/layerlist.py

```diff
@@ -67,13 +67,19 @@ def __init__(self, data=()) -> None:
         from napari._app_model.context import create_context
         from napari._app_model.context._layerlist_context import (
             LayerListContextKeys,
+            LayerListSelectionContextKeys,
         )
 
-        self._ctx = create_context(self)
-        if self._ctx is not None:  # happens during Viewer type creation
-            self._ctx_keys = LayerListContextKeys(self._ctx)
+        self._ll_ctx = create_context(self)
+        self._lls_ctx = create_context(self)
+        if self._ll_ctx is not None:  # happens during Viewer type creation
+            self._ll_ctx_keys = LayerListContextKeys(self._ll_ctx)
+        if self._lls_ctx is not None:  # happens during Viewer type creation
+            self._lls_ctx_keys = LayerListSelectionContextKeys(self._lls_ctx)
```

Comment:
First attempt of adding a context keys, to get number of layers present. I think I needed to create two `Context`s. This works but is not pretty, suggestions welcome.

I have also amended such that there are two `ContextNamespace` classes, one that takes in `layers` and one that takes in `layers.selection` (I think, not exactly really sure how it works)

cc @DragaDoncila